components/proxy/src/main/java/com/hotels/styx/admin/handlers/MetricsHandler.java

```diff
@@ -39,6 +61,61 @@
      * @param cacheExpiration duration for which generated page content should be cached
      */
     public MetricsHandler(CodaHaleMetricRegistry metricRegistry, Optional<Duration> cacheExpiration) {
-        super(checkNotNull(metricRegistry.getMetricRegistry()), cacheExpiration, new MetricsModule(SECONDS, MILLISECONDS, DO_NOT_SHOW_SAMPLES));
+        super(requireNonNull(metricRegistry.getMetricRegistry()), cacheExpiration, new MetricsModule(SECONDS, MILLISECONDS, DO_NOT_SHOW_SAMPLES));
+
+        this.metricRegistry = metricRegistry;
+    }
+
+    @Override
+    public Observable<HttpResponse> handle(HttpRequest request) {
+        return metricName(request.path())
+                .map(metricName -> specificMetricsResponse(request, metricName))
+                .orElseGet(() -> super.handle(request));
+    }
+
+    private Observable<HttpResponse> specificMetricsResponse(HttpRequest request, String metricName) {
+        Map<String, Metric> metrics = metrics(metricName);
+
+        if (metrics.isEmpty()) {
+            return just(response(NOT_FOUND).build().toStreamingResponse());
+        }
+
+        boolean pretty = request.queryParam("pretty").isPresent();
+
+        String serialised = serialise(metrics, pretty);
+
+        return just(response(OK)
+                .body(serialised, UTF_8)
+                .disableCaching()
+                .build()
+                .toStreamingResponse());
+    }
+
+    private static Optional<String> metricName(String path) {
+        Matcher matcher = SPECIFIC_METRICS_PATH_PATTERN.matcher(path);
+
+        if (matcher.matches()) {
+            return Optional.of(matcher.group(1));
+        }
+
+        return Optional.empty();
+    }
+
+    private String serialise(Object object, boolean pretty) {
+        ObjectWriter writer = pretty ? metricSerialiser.writerWithDefaultPrettyPrinter() : metricSerialiser.writer();
+
+        try {
+            return writer.writeValueAsString(object);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Map<String, Metric> metrics(String metricNameStart) {
```

Comment:
Could we rename `metricNameStart` to `metricNamePrefix` or something?

In my mind `metricNameStart` sounds like an index offset or a boolean variable.